cognite/client/cognite_client.py

```diff
@@ -112,9 +112,12 @@ def __init__(
 
         self._requests_session = self._requests_retry_session()
 
-        self._project = project
-        if project is None:
-            self._project = self.login.status().project
+        self._project = None  # Allow the LoginClient object to be created
+        self._project = self.login.status().project
+        if project is not None and project != self._project:
```

Comment:
If we choose to check login.status() no matter if they pass project or not, we may as well remove the project parameter and always fetch it using the api-key. The reasoning behind letting the user pass the project parameter was to not render the client useless in case the login endpoint is down.

Reply:
(Also that test is failing for this very reason. It passes a dummy base_url to the client, and passes project to avoid invoking login.status(). You can fix it by patching that api call.)